resolver/resolver_test.go

```diff
@@ -56,19 +67,23 @@ func TestRecurivePathResolution(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	resolver := resolver.NewBasicResolver(dagService)
-	node, err := resolver.ResolvePath(ctx, p)
+	resolver := resolver.NewBasicResolver(bsrv)
+	resolver.FetchConfig.NodeReifier = unixfsnode.Reify
+	node, lnk, err := resolver.ResolvePath(ctx, p)
 	if err != nil {
 		t.Fatal(err)
 	}
 
+	uNode, ok := node.(unixfsnode.PathedPBNode)
```

Comment:
Is UnixFS really required here, aren't we just traversing DagPb nodes?

Reply:
iiuc this go-path library is really all for unixfsv1 semantics.  If we just wanted to traverse over dagpb nodes directly, we'd... `traversal.*` over `ipld.Node`, and there'd be nothing here.

I'm hoping in the longer run, this whole library will also just disappear into being a "normal" `traversal.*` over these things from the `unixfsnode` package, which are actually ADLs that are making the unixfsv1 semantics legible as normal Node step-wise semantics.  IIUC, where were at now, with this PR, is that we're eroding go-path into being becoming this (but not getting rid of go-path yet, because incrementalism/scope-avoidance/you-know).

What this is testing is that the `NodeReifier` hook system is actually producing one of those ADLs.

Reply:
Yea what we're trying to do is maintain current behavior -- which is pathing by string, which neccesitated the unixfsnode library.

Reply:
Even with regular protonodes -- not unixfs files/directories -- you need the ADL to do pathing by string.